gpt_engineer/__init__.py

```diff
@@ -0,0 +1,17 @@
+import os
+from dotenv import load_dotenv
+
+def check_env_vars():
+    """
+    Check if API KEY env variable exist
+    If it doesn’t, load from .env file
+    """
+    if os.getenv("OPENAI_API_KEY") is None:
+        load_dotenv()
+
+        # After attempting to load from .env, check again
+        if os.getenv("OPENAI_API_KEY") is None:
+            raise ValueError("Cannot run the program without OPENAI_API_KEY environment variable.\nPlease set it in your environment or in a .env file.")
+
+# Call the function when package is imported
+check_env_vars()
```

Comment:
We have this code duplicated. According to DRY we should only have it in one place!

It's not obvious, but I thought about it and I think this is the behavior we want:
- We move the function in main.py to its own file
- We import that and run it in the main() function and in the scripts that need it
- We remove the code in __init__

In this way things are as explicit not implicit according to python philosophy.